src/main/java/org/opensearch/searchrelevance/transport/queryset/PutQuerySetTransportAction.java

```diff
@@ -72,24 +73,34 @@ protected void doExecute(Task task, PutQuerySetRequest request, ActionListener<I
     }
 
     /**
-     * Query set input is a list of queryText and referenceAnswer pair.
+     * Query set input is a list of queryText and customizedKeyValueMap pair.
+     * Converts to format: "queryText#\nkey1: value1\nkey2: value2\n..."
      * e.g:
+     * Input:
      * {
      *     "queryText": "What is OpenSearch?",
      *     "referenceAnswer": "OpenSearch is a community-driven, open source search and analytics suite"
      * }
-     * @param queryWithReferenceList - list of queryText and referenceAnswer pair
+     * Output: "What is OpenSearch?#\nreferenceAnswer: OpenSearch is a community-driven, open source search and analytics suite"
+     *
+     * @param queryWithReferenceList - list of queryText and customizedKeyValueMap pair
      * @return - querySetQueries as a list of QuerySetEntry objects
      */
     private List<QuerySetEntry> convertQuerySetQueriesList(List<QueryWithReference> queryWithReferenceList) {
         return queryWithReferenceList.stream().map(queryWithReference -> {
-            String queryText;
-            if (queryWithReference.getReferenceAnswer() != null && !queryWithReference.getReferenceAnswer().isEmpty()) {
-                queryText = String.join(DELIMITER, queryWithReference.getQueryText(), queryWithReference.getReferenceAnswer());
-            } else {
-                queryText = queryWithReference.getQueryText();
+            StringBuilder queryTextBuilder = new StringBuilder(queryWithReference.getQueryText());
+
+            // Append all key-value pairs from customizedKeyValueMap in "key: value" format
+            if (queryWithReference.getCustomizedKeyValueMap() != null && !queryWithReference.getCustomizedKeyValueMap().isEmpty()) {
+                queryTextBuilder.append(DELIMITER);
+                for (Map.Entry<String, String> entry : queryWithReference.getCustomizedKeyValueMap().entrySet()) {
+                    if (entry.getValue() != null && !entry.getValue().isEmpty()) {
+                        queryTextBuilder.append("\n").append(entry.getKey()).append(": ").append(entry.getValue());
```

Comment:
We need to rethink this format later, that's too fragile, those delimiter characters can be part of the user provided payload. Something like json format is more stable